Does not type check if withTypeProvider is not fluent

Consider the following:

```typescript
import Fastify from 'fastify';
import { TypeBoxTypeProvider } from '@fastify/type-provider-typebox';
import { Type } from '@sinclair/typebox';

const server = Fastify();

server.withTypeProvider<TypeBoxTypeProvider>();

server.post('/', {
  schema: {
    body: Type.Object({
      name: Type.String()
    })
  }
}, async (request, reply) => {
  const { name } = request.body;
  return { hello: name };
})

server.listen({ port: 3000 })

```

This does not work but it errors with:

```
server.ts:16:11 - error TS2339: Property 'name' does not exist on type 'unknown'.

16   const { name } = request.body;
             ~~~~
```

However if we inline the route definition and `withTypeProvider` call it works:

```typescript
import Fastify from 'fastify';
import { TypeBoxTypeProvider } from '@fastify/type-provider-typebox';
import { Type } from '@sinclair/typebox';

const server = Fastify();

server.withTypeProvider<TypeBoxTypeProvider>().post('/', {
  schema: {
    body: Type.Object({
      name: Type.String()
    })
  }
}, async (request) => {
  const { name } = request.body;
  return { hello: name };
})

server.listen({ port: 3000 })
```

--- a/README.md
+++ b/README.md
@@ -17,6 +17,17 @@ import { TypeBoxTypeProvider } from '@fastify/type-provider-typebox'
 const fastify = Fastify().withTypeProvider<TypeBoxTypeProvider>()
 ```
 
+Note that the following will not work:
+
+```ts
+import Fastify from 'fastify'
+import { TypeBoxTypeProvider } from '@fastify/type-provider-typebox'
+
+const fastify = Fastify()
+
+fastify.withTypeProvider<TypeBoxTypeProvider>()
+```
+
 ## Example
 
 ```ts

